packages/assets-controllers/src/MultichainAssetsController/utils.ts

```diff
@@ -0,0 +1,32 @@
+import type {
+  Caveat,
+  PermissionConstraint,
+} from '@metamask/permission-controller';
+import { SnapCaveatType } from '@metamask/snaps-utils';
+
+// TODO: this is a duplicate of https://github.com/MetaMask/snaps/blob/362208e725db18baed550ade99087d44e7b537ed/packages/snaps-rpc-methods/src/endowments/name-lookup.ts#L151
+// To be removed once core has snaps-rpc-methods dependency
+/**
+ * Getter function to get the chainIds caveat from a permission.
+ *
+ * This does basic validation of the caveat, but does not validate the type or
+ * value of the namespaces object itself, as this is handled by the
+ * `PermissionsController` when the permission is requested.
+ *
+ * @param permission - The permission to get the `chainIds` caveat from.
+ * @returns An array of `chainIds` that the snap supports.
+ */
+// istanbul ignore next
```

Comment:
For other reviews: Since this function comes from the `snaps-*` packages, we decided to ignore the coverage for this one.